Update README.md (#1173)
eth-cscs/DLA-Future: README.md

# Distributed Linear Algebra from the Future

> DLA-Future is a distributed linear algebra library implemented using C++ proposal P2300 which provides:

Reviewer: I think we can do this now that it's been voted in:
```suggestion
DLA-Future is a distributed linear algebra library implemented using C++ `std::execution` (possibly link to P2300 paper?) which provides:
```

> DLA-Future is a distributed linear algebra library implemented using C++ proposal P2300 which provides:

Reviewer 2: ```suggestion
DLA-Future is a distributed linear algebra library implemented using C++ proposal [P2300](https://cplusplus.github.io/sender-receiver/execution.html) (`std::execution`) which provides:
```

- An asynchronous C++ interface,
- A synchronous C interface,
- A synchronous ScaLAPACK-like C interface (ScaLAPACK drop-in with a subset of ScaLAPACK arguments. E.g. workspace arguments are not present),
- A synchronous ScaLAPACK-like Fortran interface (See [DLA-Future-Fortran](https://github.com/eth-cscs/DLA-Future-Fortran)).

> 

Reviewer: Maybe here, or up in the first sentence:
`DLA-Future runs on x86 and ARM CPUs as well as NVIDIA and AMD GPUs.`?

> Currently it provides the following algorithms :

Reviewer 2: ```suggestion
Currently the asynchronous C++ interface provides the following algorithms :
```

> Currently it provides the following algorithms :

Reviewer 3: ```suggestion
Currently it provides the following algorithms:
```

- Matrix Multiplication:
  - Triangular (`P[S,D,C,Z]TRMM`)
  - Hermitian (`P[C,Z]HEMM` and `P[S,D]SYMM`)
- Triangular Solver (`P[S,D,C,Z]TRSM`)
- Cholesky Decomposition

> - Symmetric / Hermitian eigenvalue solver (`P[C,Z]HEEVD` and `P[S,D]SYEVD`. Can replace any call to ScaLAPACK`P[C,Z]HEEV*` and `P[S,D]SYEV*`.)

Reviewer 3: ?

```suggestion
- Symmetric / Hermitian eigenvalue solver (`P[C,Z]HEEVD` and `P[S,D]SYEVD`. It can replace any call to ScaLAPACK`P[C,Z]HEEV*` and `P[S,D]SYEV*`).
```

Author: Outdated

> - Generalized Symmetric / Hermitian eigenvalue solver (`P[C,Z]HEGVD` and `P[S,D]SYGVD`.  Can replace any call to ScaLAPACK `P[C,Z]HEGV*` and `P[S,D]SYGV*`.)

Reviewer 3: ?

```suggestion
- Generalized Symmetric / Hermitian eigenvalue solver (`P[C,Z]HEGVD` and `P[S,D]SYGVD`. It can replace any call to ScaLAPACK `P[C,Z]HEGV*` and `P[S,D]SYGV*`).
```

Author: Outdated

Notes:

> - C++ interface provides a non distributed interface as well.

Reviewer: ```suggestion
- The C++ interface provides a non distributed interface as well.
```

> - Eigensolvers which computes only a part of the eigenspectrum will be added in future releases.

Reviewer: Can we leave this out?

# Distributed Linear Algebra from the Future

> DLA-Future is a distributed linear algebra library implemented using using C++ `std::execution` [P2300](https://cplusplus.github.io/sender-receiver/execution.html) which provides:

Reviewer 2: ```suggestion
DLA-Future is a distributed linear algebra library implemented using C++ `std::execution` [P2300](https://cplusplus.github.io/sender-receiver/execution.html) which provides:
```

> - An asynchronous C++ interface,
> - A synchronous C interface,
> - A synchronous ScaLAPACK-like C interface (ScaLAPACK drop-in with a subset of ScaLAPACK arguments. E.g. workspace arguments are not present),
> - A synchronous ScaLAPACK-like Fortran interface (See [DLA-Future-Fortran](https://github.com/eth-cscs/DLA-Future-Fortran)).

Reviewer 2: ```suggestion
- An asynchronous C++ interface,
- a synchronous C interface,
- a synchronous ScaLAPACK-like C interface (ScaLAPACK drop-in with a subset of ScaLAPACK arguments. E.g. workspace arguments are not present),
- a synchronous ScaLAPACK-like Fortran interface (See [DLA-Future-Fortran](https://github.com/eth-cscs/DLA-Future-Fortran)).
```
? Since we are using punctuation in the bullet list.
